packages/components/progress-bar/src/progress-bar.styles.tsx

```diff
@@ -0,0 +1,62 @@
+import { css, keyframes } from '@emotion/react';
+import { designTokens } from '@commercetools-uikit/design-system';
+import { getMaxPropTokenValue } from '../helpers';
+import { type TProgressBarProps } from './progress-bar';
+
+export const heightPerScale = {
+  '10': designTokens.spacing25,
+  '20': designTokens.spacing40,
+};
+
+const progressPulse = keyframes`
+  0% {
+    background-position: 100% 0;
+  }
+  100% {
+    background-position: -100% 0;
+  }
+`;
+
+export const getLabelStyles = (
+  props: TProgressBarProps & { textAlignment: string }
+) => css`
+  min-height: ${heightPerScale[props.height ?? '20']};
+  text-align: ${props.textAlignment};
+  max-width: ${getMaxPropTokenValue(props.labelWidth!)};
+  display: inline-flex;
+  > span {
+    width: max-content;
+  }
+`;
```

Comment:
@CarlosCortizasCT A solution for not wrapping the label text unless the label text actually exceeds the width of the parent. WDYT?

Reply:
We're getting close, but there's an issue with some extra horizontal space:

![image](https://github.com/commercetools/ui-kit/assets/97907068/0d71bbc1-4d93-4048-8d61-6251f14a6979)

As you can see, we have some initial extra space we don't want to have.
Think about a use case where we want to render the progress bar beside another element: it would work well if the label text fits in one line, but if there are more, then we have this unwanted extra space.

I'm trying to find a solution for that, but no luck so far.

Reply:
I'm still struggling to find a solution for this one.

I'll ask for help to my team on Monday.

Reply:
Moving `max-width` to the child `span` element seems to work but still would require the consumer to pay close attention to their label width choice. 
```
export const getLabelStyles = (
  props: TProgressBarProps & { textAlignment: string }
) => css`
  min-height: ${heightPerScale[props.height ?? '20']};
  text-align: ${props.textAlignment};
  display: inline-flex;
  > span {
    width: max-content;
    max-width: ${getMaxPropTokenValue(props.labelWidth!)};
  }
`;
```

Reply:
Thanks @tylermorrisford 

I still see the same problem.
If you don't can you please share a similar screenshot?

![Screenshot 2024-02-05 at 09 52 46](https://github.com/commercetools/ui-kit/assets/97907068/b4fa7b82-e503-4037-a21a-81a1b4c8e42d)

Reply:
I'm still seeing it as well.

Reply:
We've been trying to find a solution to this problem but were not able to, so we weighed to options:
1. keep trying to find a solution
2. accept this as an edge case where we won't have the exact styles required

We've discussed this options with @FilPob and come to an agreement to accept the behaviour as it is.
We don't expect consumers to use the component in a way such it will reproduce the edge case.

If we ever find an important consumer use case where this is not acceptable, we can come back and research again for a solution.